org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/GradleDistribution.java

```diff
@@ -81,4 +84,50 @@ public static FixedVersionGradleDistribution forVersion(String version) {
     public static WrapperGradleDistribution fromBuild() {
         return WRAPPER_DISTRIBUTION;
     }
+
+    /**
+     * Deserializes a distribution.
+     * <p>
+     * This method recognizes strings returned by {@link #toString()}.
+     *
+     * @param distributionString the string representation of a distribution
+     * @return the parsed distribution
+     * @throws RuntimeException if the deserialization fails
+     */
+    public static GradleDistribution fromString(String distributionString) {
```

Comment:
I think our general code style is to have static members first.

Reply:
Rearranged.